lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -854,30 +860,34 @@ namespace Js
             // matching scriptContext might have been deleted and the javascriptLibrary->scriptContext
             // field reset (but javascriptLibrary is still alive).
             // Use the recycler field off library instead of scriptcontext to avoid av.
+
+            // Recycler may not be available at Dispose. We need to
+            // free the memory and report that it has been freed at the same
+            // time. Otherwise, AllocationPolicyManager is unable to provide correct feedback
+#if _WIN64
+            //AsmJS Virtual Free
+            //TOD - see if isBufferCleared need to be added for free too
+            if (IsValidVirtualBufferLength(this->bufferLength) && !isBufferCleared)
+            {
+              LPVOID startBuffer = (LPVOID)((uint64)buffer);
+              BOOL fSuccess = VirtualFree((LPVOID)startBuffer, 0, MEM_RELEASE);
+              Assert(fSuccess);
+              isBufferCleared = true;
+            }
+            else
+            {
+              free(buffer);
+            }
+#else
+            free(buffer);
+#endif
+
             Recycler* recycler = GetType()->GetLibrary()->GetRecycler();
```

Comment:
This is arraybuffer. The virtual buffer (via heapalloc or CoTaskMemAlloc) can be accessed from one ArrayBuffer. You need to neuter it before give it to another thread.